SignInCheckIn/MinistryPlatform.Translation/Repositories/EventRepository.cs

```diff
@@ -118,10 +119,30 @@ public void ImportEventSetup(string authenticationToken, int destinationEventId,
                 .PostStoredProc(ImportEventStoredProcedureName, new Dictionary<string, object> {{"@DestinationEventId", destinationEventId}, {"@SourceEventId", sourceEventId}});
         }
 
-        public List<MpEventDto> GetEventAndCheckinSubevents(string token, int eventId)
+        public List<MpEventDto> GetEventAndCheckinSubevents(string authenticationToken, int eventId)
         {
+            // JPC - check to see why this was null
+            var token = authenticationToken ?? _apiUserRepository.GetToken();
+
             return _ministryPlatformRestRepository.UsingAuthenticationToken(token)
                 .Search<MpEventDto>($"(Events.Event_ID = {eventId} OR Events.Parent_Event_ID = {eventId}) AND Events.[Allow_Check-in] = 1", _eventColumns);
         }
+
+	public List<MpEventDto> GetSubeventsForEvents(List<int> eventIds, int? eventTypeId)
+        {
+            var apiUserToken = _apiUserRepository.GetToken();
+
+            var queryString = eventIds.Aggregate("(", (current, id) => current + (id + ","));
+
+            queryString = queryString.TrimEnd(',');
+            queryString += ")";
+
+            // search on the event type if it's not a null param
+            var typeQueryString = " AND Events.[Event_Type_ID] = " + eventTypeId;
+
```

Comment:
since eventTypeId is optional, will this query still work if it's not passed in? Seems like it would just add on ` AND Events.[Event_Type_ID] = ` to the end of the query if not